### ISSUE TITLE
Do I understand structure of  RNN correctly?

### ISSUE THREAD

**Issue by Yingyingzhang15** [fchollet/keras#2074]
Hi. I feel confused about the structure of rnn.My model is as follows:

tsteps = 50
dropout=0.5
model = Sequential()
model.add(LSTM(150, input_shape=(tsteps, 3)))
model.add(Dropout(dropout))
model.add(Dense(2))
model.add(Activation('softmax'))
model.compile(loss='binary_crossentropy', optimizer='adam')

My understanding of this model is:
The input data size is 3d (x1,x2,x3), and the model take every tsteps 3d vector as a whole.
There are two layers in this model.
The LSTM layer has 150 neurons, so the output dimension is 150.
The dense layer has 3 nerons, so the output dimension is 3.
Do I understand it correctly?Thank you in advance!

**Comment by ymcui** [comment]
yes, you are right.
the LSTM receives 3D tensor shape of (batch_size, timestep, input_dim) and output a 2D tensor with shape of (batch_size, output_dim)
so, in your example, your input should be in (batch_size, tsteps, 3), and output should be in (batch_size, 2)

**Comment by Yingyingzhang15** [comment]
@ymcui Thank you very much!:)

**Comment by Yingyingzhang15** [comment]
@ymcui Does my model here #2074 have two layers(LSTM layer and dense layer ) or three layers (dense layer, LSTM layer and dense layer)?

The input_dim is 3, and I don't know if it is a layer with 3 neurons. 

Thank you !

**Comment by ymcui** [comment]
actually, you have 4 layers
```
model.add(LSTM(150, input_shape=(tsteps, 3)))
model.add(Dropout(dropout))
model.add(Dense(2))
model.add(Activation('softmax'))
```
the last two lines can be merged into
`mode.add(Dense(2,activation='softmax'))`

>The input_dim is 3, and I don't know if it is a layer with 3 neurons.

yes, you should prepare you input with shape of (50,3), where 50 is the length of each sequence, and 3 is the dimension of each element in sequence.

**Comment by Yingyingzhang15** [comment]
@ymcui Thank you very much for your help!
